### PR TITLE
Add a template for the new link component

### PR DESCRIPTION
**Pull request by msayoung**

### CODE REVIEW

**Comment by finnlewis** [comment]
@markconroy another one that @msayoung would like your eyes on before merging. :)

File: css/components/link.css [hunk 1]
```diff
@@ -0,0 +1,14 @@
+.link-block__title-wrapper {
+  display: flex;
+  align-items: center;
+  margin-bottom: var(--vertical-rhythm-spacing);
+}
+
+.link-block__title-icon {
+  flex-shrink: 0;
+  margin-right: var(--spacing-smaller);
```
Comment (markconroy) [inline css/components/link.css]: We should probably create variables for these margins, so it's easier for sub-themes to override them

File: css/components/link.ie11.css [hunk 2]
```diff
@@ -0,0 +1,7 @@
+.link-block__title-wrapper {
+  margin-bottom: 1.5rem;
+}
+
+.link-block__title-icon {
+  margin-right: 0.5rem;
```
Comment (markconroy) [inline css/components/link.ie11.css]: Let's use variables for these margins

File: templates/paragraphs/paragraph--localgov-link.html copy.twig [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+{% set title_icon = 'chevron-right' %}
+
+<div{{ attributes.addClass(classes) }}>
+
+  {% if paragraph.localgov_url %}
+    <div class="link-block__title-wrapper">
+      {% include "@localgov_base/includes/icons/icon.html.twig" with {
+          icon_name: title_icon,
+          icon_wrapper_element: 'div',
```
Comment (markconroy) [inline templates/paragraphs/paragraph--localgov-link.html copy.twig]: I _think_ `div` is the "sensible default", so does not need to be included here as it is set automatically.

File: templates/paragraphs/paragraph--localgov-link.html copy.twig [hunk 4]
```diff
@@ -0,0 +1,80 @@
+{#
```
Comment (markconroy) [inline templates/paragraphs/paragraph--localgov-link.html copy.twig]: Looks like this file is a duplicate

File: css/components/link.css (hunk 1 above)
Comment (msayoung) [reply css/components/link.css]: I realised we didn't need that margin-bottom anyway

File: templates/paragraphs/paragraph--localgov-link.html copy.twig (hunk 3 above)
Comment (msayoung) [reply templates/paragraphs/paragraph--localgov-link.html copy.twig]: do you mean the sensible default for the icon? 
if so then this div is wrapping the icon and the text and flexing them. but I may have misunderstood :)

File: templates/paragraphs/paragraph--localgov-link.html copy.twig (hunk 4 above)
Comment (msayoung) [reply templates/paragraphs/paragraph--localgov-link.html copy.twig]: gone

File: templates/paragraphs/paragraph--localgov-link.html copy.twig (hunk 3 above)
Comment (markconroy) [reply templates/paragraphs/paragraph--localgov-link.html copy.twig]: I mean the div on line 64 `icon_wrapper_element: 'div',` - the icon template uses `div` by default, so you only need to put a value in there is you want to use `span` or something elsse.

File: css/components/link.css (hunk 1 above)
Comment (markconroy) [reply css/components/link.css]: Should we create a variable for `margin-right` here? Something like `--link-title-icon-margin`?

**Pull request merged**